### PR TITLE
Fix pull image: check response code

### PR DESCRIPTION
**Pull request by jimmyxian**
Fix: https://github.com/docker/swarm/issues/927
Signed-off-by: Xian Chaobo <xianchaobo@huawei.com>

### CODE REVIEW

**Comment by jimmyxian** [comment]
ping @vieux @aluzzardi @ehazlett

File: dockerclient.go [hunk 1]
```diff
@@ -453,7 +453,20 @@ func (client *DockerClient) PullImage(name string, auth *AuthConfig) error {
 	if err != nil {
 		return err
 	}
+
 	defer resp.Body.Close()
+	if resp.StatusCode == 404 {
+		return ErrNotFound
+	}
+	if resp.StatusCode >= 400 {
+		defer resp.Body.Close()
```
Comment (donhcd) [inline dockerclient.go]: there's already a defer on 457

Comment (jimmyxian) [inline dockerclient.go]: OK, fix right now.

**Comment by jimmyxian** [comment]
ping @donhcd Check agian. thanks

**Comment by donhcd** [comment]
@jimmyxian  do you have an example of the error in the second case (status != 404 && status >= 400)? I'm wondering if it's better or worse than [the existing error handling](https://github.com/jimmyxian/dockerclient/commit/6d0468a232b55dc1fd2dc0b15d8352896dfa13e9#diff-a07fd1daa03a253fbbe16804edefc0d8R476)

**Comment by jimmyxian** [comment]
@donhcd Yeah
Use client , Do
`client.PullImage("4e8aa3148a132f19ec560952231c4d39522043994df7d2dc239942c0f9424ebd", nil)`

**Comment by donhcd** [comment]
okay, I played around with running this on 1.6.1 and I'm getting

| action | status code | error format |
| -------- | ---- | ------------------------ |
|pulling a nonexistent image | 200 | JSON |
|pulling an image ID | 500 | string (not JSON) |

so I guess this PR handles that.

LGTM

**Comment by ehazlett** [comment]
LGTM

**Pull request merged**